backend/src/AppServer.hs

```diff
@@ -77,53 +104,70 @@ readOrError errMsg action =
       logErrorN errMsg
       throwError err500 {errBody = show errMsg}
 
-submitReportHandler :: RawGameReport -> AppM SubmitGameReportResponse
```

Comment:
This function has been blown to pieces. Perhaps start with its new definition, and work from there?